### ISSUE TITLE
Issues with Multiple Connections

### ISSUE THREAD

**Issue by CCS-StevenPretlove** [ecitsolutions/Autotask#53]
Trying to write a script with multiple connections however i keep get "incorrect user / password" for the second connection

if i copy out the connection details and then paste into another powershell session / script they work fine.

always halts at 

VERBOSE: Connect-AtwsWebServices: Creating New-WebServiceProxy against URI: https://webservices16.autotask.net/ATServices/1.6/atws.wsdl
Import-Module : Could not connect to Autotask WebAPI. Verify your credentials. If you are sure you have the rights - maybe you typed your password wrong?

any thoughts ?

**Comment by klemmestad** [comment]
Unfortunately, yes, this does not work even though I have claimed it would. To interact with Autotask I use `New-WebServiceProxy -Namespace Autotask ...` to import all classes and make them available under the Autotask. prefix (`New-Object Autotask.Account`). Unfortunately this can only be done once pr named namespace. When you try to add a second connection .NET seems to reuse the previously created dynamic library, effectively blocking you from creating a new one.

The solution is to use a different namespace for the second connection. Unfortunately the current namespace name is hardcoded a ton of places (I did not see this one coming at all) and changing it will probably break a ton of things.

I still plan to do it 😊 My first attempt was to connect it to the -Prefix functionality of `Import-Module`. So far I have not figured out how to detect -Prefix from within the module loading script (.psm1). As far as I can tell Import-Module just doesn’t pass it along to the module code. So I will have to add another parameter to the load process. So it will come to a module near you, however I have a few other tweaks and bugfixes I need to get out the door first 😊
